aea/protocols/base.py

```diff
@@ -81,6 +81,22 @@ def __init__(self, _body: Optional[Dict] = None, **kwargs):
         except Exception as e:  # pylint: disable=broad-except
             _default_logger.error(e)
 
+    def to_json(self) -> dict:
```

Comment:
just use existing encode and decode logic and create a generic JsonSerializer please

Reply:
`from_json` logic can not be ported to generic serializer decode, cause it requires a class of a Message type to restore a message.

Reply:
updated with base64 and message.encode